src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs

```diff
@@ -361,5 +362,93 @@ private static FixAllSuggestedAction GetFixAllSuggestedAction(IEnumerable<Sugges
 
         public void MessageBox(string message)
             => ExecuteOnActiveView(view =>  System.Windows.MessageBox.Show(message));
+
+        public void VerifyDialog(string dialogAutomationId, bool isOpen)
+        {
```

Comment:
All automationelement work can be done from a background thread. The only reason this even needs to be in-proc is to get the VS hwnd from DTE.